crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -371,8 +375,38 @@ impl ScheduleGraph {
     }
 
     fn add_systems<P>(&mut self, systems: impl IntoSystemConfigs<P>) {
-        let SystemConfigs { systems, chained } = systems.into_configs();
-        let mut system_iter = systems.into_iter();
+        let SystemConfigs {
+            systems,
+            graph_info,
+            conditions,
+            chained,
+        } = systems.into_configs();
+        if conditions.is_empty()
```

Comment:
I think its worth moving these checks into a `should_add_anonymous_set` method on SystemConfigs, with a comment explaining the purpose. We should also add a comment to the SystemConfigs struct explaining that if new fields are added, they should be considered for inclusion in `should_add_anonymous_set`.

Reply:
Done.